caddyhttp/proxy/proxy_test.go

```diff
@@ -1378,11 +1403,29 @@ func (r *noopReader) Read(b []byte) (int, error) {
 func newFakeUpstream(name string, insecure bool) *fakeUpstream {
 	uri, _ := url.Parse(name)
 	u := &fakeUpstream{
-		name: name,
-		from: "/",
+		name:    name,
+		from:    "/",
+		timeout: 30 * time.Second,
 		host: &UpstreamHost{
 			Name:         name,
-			ReverseProxy: NewSingleHostReverseProxy(uri, "", http.DefaultMaxIdleConnsPerHost),
+			ReverseProxy: NewSingleHostReverseProxy(uri, "", http.DefaultMaxIdleConnsPerHost, 30*time.Second),
+		},
+	}
+	if insecure {
+		u.host.ReverseProxy.UseInsecureTransport()
+	}
+	return u
+}
+
+func newFakeUpstreamWithTimeout(name string, insecure bool, timeout time.Duration) *fakeUpstream {
```

Comment:
This is duplicate of `newFakeUpstream` except for the timeout param. It's better if you supply a timeout parameter to `newFakeUpstream` and use that value or the default.

Reply:
Is the approach in the latest commit acceptable?
I do not want to make timeout a dependency of creating an upstream because that will mean changing the function call in all the callers. 
This change should remove duplication and serve as the default value supplier in callers that choose not to provide a timeout.

Reply:
Thanks for the input. I can refactor them. I was not quite sure on the correct approach, but what you say makes sense.
I'll make the changes.